packages/devtools_app/lib/src/screens/inspector_v2/inspector_tree_controller.dart

```diff
@@ -1128,37 +1133,57 @@ class _RowPainter extends CustomPainter {
 
   @override
   void paint(Canvas canvas, Size size) {
-    double currentX = 0;
     final paint = _defaultPaint(colorScheme);
 
     final InspectorTreeNode node = row.node;
     final bool showExpandCollapse = node.showExpandCollapse;
+    final distanceFromExpandCollapseToRowStart =
+        inspectorColumnIndent * expandCollapseToRowStartDistancePercentage;
     for (final tick in row.ticks) {
-      currentX = _controller.getDepthIndent(tick) - inspectorColumnWidth * 0.5;
+      final expandCollapseX = _controller.getDepthIndent(tick) -
+          distanceFromExpandCollapseToRowStart;
       // Draw a vertical line for each tick identifying a connection between
       // an ancestor of this node and some other node in the tree.
       canvas.drawLine(
-        Offset(currentX, 0.0),
-        Offset(currentX, inspectorRowHeight),
+        Offset(expandCollapseX, 0.0),
+        Offset(expandCollapseX, inspectorRowHeight),
         paint,
       );
     }
     // If this row is itself connected to a parent then draw the L shaped line
     // to make that connection.
     if (row.lineToParent) {
-      currentX = _controller.getDepthIndent(row.depth - 1) -
-          inspectorColumnWidth * 0.5;
+      final parentExpandCollapseX = _controller.getDepthIndent(row.depth - 1) -
+          distanceFromExpandCollapseToRowStart;
       final double width = showExpandCollapse
-          ? inspectorColumnWidth * 0.5
-          : inspectorColumnWidth;
+          ? inspectorColumnIndent * 0.6
+          : inspectorColumnIndent;
+      canvas.drawLine(
+        Offset(parentExpandCollapseX, 0.0),
+        Offset(parentExpandCollapseX, inspectorRowHeight * 0.5),
+        paint,
+      );
       canvas.drawLine(
-        Offset(currentX, 0.0),
-        Offset(currentX, inspectorRowHeight * 0.5),
+        Offset(parentExpandCollapseX, inspectorRowHeight * 0.5),
+        Offset(parentExpandCollapseX + width, inspectorRowHeight * 0.5),
         paint,
       );
+    }
+
+    if (row.hasSingleChild && node.isExpanded) {
+      final distanceFromIconCenterToRowStart =
+          inspectorColumnIndent * iconCenterToRowStartDistancePercentage;
+      final iconCenterX = _controller.getDepthIndent(row.depth) -
+          distanceFromIconCenterToRowStart;
       canvas.drawLine(
-        Offset(currentX, inspectorRowHeight * 0.5),
-        Offset(currentX + width, inspectorRowHeight * 0.5),
+        Offset(
```

Comment:
can you add a comment explaining the math behind 0.75 and 1.25?

Reply:
Added a comment and pulled the percentages into const variables as well